### ISSUE TITLE
missing input type = time

### ISSUE THREAD

**Issue by hans2103** [joomla/joomla-cms#22140]
### Is your feature request related to a problem? Please describe.
For my component Example I define the form fields in `components/com_example/models/forms/example.xml`

```
<field name="time"
	   type="time"
	   label="COM_EXAMPLE_TIME_LABEL"
/>
```

This will be rendered like:
```
<input type="text" name="time" />
```

I was expecting to see
```
<input type="time" name="time" />
```

After searching in Joomla code (`libraries/joomla/form/fields` and `libraries/src/Form`) I've found out that type=time does not exist yet in our codebase.

### Describe the solution you'd like
It would be great if it is possible to use type=time in Joomla.
> `<input>` elements of type **time** create input fields designed to let the user easily enter a time (hours and minutes, and optionally seconds).
> 
> The control's user interface will vary from browser to browser. Support is good in modern browsers, with Safari being the sole major browser not yet implementing it; in Safari, and any other browsers that don't support `<time>`, it degrades gracefully to `<input type="text">`.
text copied from: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/time

First step was to create this issue, next step is to create a PR. :-)

### Additional context
More information about input type=time can be found on w3.org
https://www.w3.org/TR/2012/WD-html-markup-20120315/input.time.html

**Comment by brianteeman** [comment]
If it is for J4 we can use the html5 patterns

**Comment by Fedik** [comment]
for note: it also require to do fallback script, 
because it has bad support by a browsers  https://caniuse.com/#feat=input-datetime

**Comment by hans2103** [comment]
@Fedik no coded fallback needed. As mentioned in the quote from mozilla.org it will gracefully degrades to `<input type="text">` on Safari and other browsers that don't support `<time>`.
When I look at caniuse I see 86.66 % Global support. I cannot say that it means _bad support_

**Comment by Fedik** [comment]
> As mentioned in the quote from mozilla.org it will gracefully degrades to <input type="text"> on Safari and other browsers that don't support <time>

this you will try to explain for further issue reports: "the time field do not display time input", or "time input ignore min/max attribute" :wink:  

but personally I do not mind, I just point on potential issue

**Comment by ghost** [comment]
New Features go only in j4 as i know, so this should get a J4-Issue-Label.

**Comment by ghost** [comment]
@infograf768 can you please add j4-Label?

**Comment by VladikB** [comment]
I am working on this. Which files should be changed to add a new field type?

**Comment by Bakual** [comment]
@VladikB You would have to add a new file named `TimeField.php` here: https://github.com/joomla/joomla-cms/tree/4.0-dev/libraries/src/Form/Field
You can have a look at the other fields to see how they work.

**Comment by VladikB** [comment]
Okay, thanks, got it. And I how do I get the min, max and step attributes to work and checkt them out?

**Comment by zero-24** [comment]
Take a look here: https://github.com/joomla/joomla-cms/blob/4.0-dev/libraries/src/Form/Field/NumberField.php there we also support min, max and step :)

**Comment by ghost** [comment]
Closed as having Pull Request #26184
